### PR TITLE
emacs: make gnutls dependency required instead of optional

### PR DESCRIPTION
**Pull request by raxod502**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

If Emacs is not built with GnuTLS, then it cannot make HTTPS connections, meaning it is impossible to install packages without installing workarounds. This is undesirable default behavior.

Thoughts on changing the default to fix this problem, so that Emacs works out of the box?

### CODE REVIEW

**Comment by ilovezfs** [comment]
@zmwangx any thoughts here?

**Comment by DomT4** [comment]
> Can it use depends_on "openssl"?

IIRC, I think support is there for that.

**Comment by raxod502** [comment]
I didn't find any flags in Emacs' `./configure` relating to SSL, other than for GnuTLS.

I also didn't find anything promising from a Google search for "emacs openssl".

If it's possible for Emacs to use OpenSSL instead of GnuTLS, it would seem to be by customizing `tls-program` from its default value of

```elisp
  '("gnutls-cli --x509cafile %t -p %p %h"
    "gnutls-cli --x509cafile %t -p %p %h --protocols ssl3")
```

to something which includes an OpenSSL command. That would require Homebrew to insert something into `site-start.el`, or things wouldn't work out of the box. Further, the above values of `tls-program` don't work out of the box either on macOS, unless `libressl` is installed and `gnutls-trustfiles` is customized to include `/usr/local/etc/libressl/cert.pem`. The same caveat might apply to OpenSSL, although I don't know if that's the case.

In summary, using OpenSSL does not seem to me like a viable alternative to GnuTLS. Perhaps I'm missing something, though.

**Comment by raxod502** [comment]
Possibly relevant: https://github.com/skeeto/elfeed/issues/41

**Comment by ilovezfs** [comment]
@dunn any thoughts on this change?

**Comment by dunn** [comment]
Fine with me!

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @raxod502! Without people like you submitting PRs we couldn't run this project. You're awesome!
